### PR TITLE
Update UI to account for naming assessment types within the scope of …

### PR DESCRIPTION
**Pull request by wtritch**
…a subject.

image

image

image

image

### CODE REVIEW

File: webapp/src/main/webapp/src/app/csv-export/csv-builder.service.ts [hunk 1]
```diff
@@ -145,7 +145,10 @@ export class CsvBuilder {
   withAssessmentType(getAssessment: (item: any) => Assessment) {
     return this.withColumn(
       this.translateService.instant('csv-builder.assessment-type'),
-      (item) => this.translateService.instant(`common.assessment-type.${getAssessment(item).type}.short-name`)
+      (item) => {
+        const assessment: Assessment = getAssessment(item);
+        return this.translateService.instant(`subject.${assessment.subject}.asmt-type.${assessment.type}.name`);
+      }
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/csv-export/csv-builder.service.ts]: Updated the CSV export to pick the assessment type name within the context of a subject.

File: webapp/src/main/webapp/src/app/shared/i18n/rdw-translate-loader.ts [hunk 2]
```diff
@@ -8,27 +8,33 @@ import { HttpClient } from "@angular/common/http";
...
       this.getClientTranslations(languageCode),
-      this.getServerTranslations(languageCode)
+      this.getServerTranslations(languageCode),
+      this.subjectService.getSubjectCodes()
     ).pipe(
-      map(([ clientTranslations, serverTranslations ]) => {
-        return _.merge(clientTranslations, serverTranslations);
+      map(([ clientTranslations, serverTranslations, subjects ]) => {
+        const asmtTranslations = this.createSubjectTranslations(subjects, serverTranslations);
+        return _.merge(clientTranslations, asmtTranslations, serverTranslations);
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/shared/i18n/rdw-translate-loader.ts]: This just injects/overrides the `common.assessment-type.[TYPE].short-name` values to the combined assessment type labels.

Comment (jtreuting) [reply webapp/src/main/webapp/src/app/shared/i18n/rdw-translate-loader.ts]: clever

Comment (wtritch) [reply webapp/src/main/webapp/src/app/shared/i18n/rdw-translate-loader.ts]: https://www.youtube.com/watch?v=_O1hM-k3aUY

**Pull request merged**